workers/tests/integrationtests/test_api_report_geojson_io.py

```diff
@@ -39,30 +36,21 @@ def setUp(self):
         self.feature_id = 3
         self.fid_field = "ogc_fid"
 
-        self.response_schema = get_response_schema()
-        self.feature_schema = get_feature_schema()
-        self.featurecollection_schema = get_featurecollection_schema()
-
         report = SimpleReport()
         report.set_indicator_layer()
-        self.number_of_indicators = len(report.indicator_layer)
+        number_of_indicators = len(report.indicator_layer)
+
+        self.general_schema = get_general_schema()
+        self.feature_schema = get_report_feature_schema(number_of_indicators)
+        self.featurecollection_schema = get_featurecollection_schema()
 
     def run_tests(self, response, featurecollection=False) -> None:
         self.assertEqual(response.status_code, 200)
-        response_json = response.json()
-        self.validate(response_json, self.response_schema)
-        if featurecollection is False:
-            features = [response_json]
-        else:
-            features = response_json["features"]
-        for feature in features:
-            for i in range(0, self.number_of_indicators):
-                feature_schema = get_feature_schema(i)
-                self.validate(feature, feature_schema)
-
-    def validate(self, geojson: dict, schema: Schema) -> None:
-        schema.validate(geojson)  # Print information if validation fails
-        self.assertTrue(schema.is_valid(geojson))
+
+        response_content = geojson.loads(response.content)
+        self.assertTrue(response_content.is_valid)  # Valid GeoJSON?
+        self.assertTrue(self.general_schema.is_valid(response_content))
+        self.assertTrue(self.feature_schema.is_valid(response_content))
```

Comment:
is the `feature_schema` handling having more indicators than defined?

Reply:
In this case `self.feature_schema` is derived from calling `get_report_feature_schema(number_of_indicators)`. If the response contains more indicators then the `numbers_of_indicators` passed to `get_report_feature_schema` the validity check will fail.

Does this answer your question?

Reply:
yes, thanks